data-prepper-api/src/main/java/org/opensearch/dataprepper/model/codec/InputCodec.java

```diff
@@ -31,8 +32,19 @@ public interface InputCodec {
      * @param eventConsumer The consumer which handles each event from the stream
      * @throws IOException throws IOException when invalid input is received or incorrect codec name is provided
      */
-    void parse(
+    default void parse(
             InputFile inputFile,
             DecompressionEngine decompressionEngine,
-            Consumer<Record<Event>> eventConsumer) throws IOException;
+            Consumer<Record<Event>> eventConsumer) throws IOException {
+        Objects.requireNonNull(inputFile);
+        Objects.requireNonNull(eventConsumer);
+        InputStream inputStream = inputFile.newStream();
+        try {
+            parse(decompressionEngine.createInputStream(inputStream), eventConsumer);
+            inputStream.close();
+        } catch (Exception e) {
```

Comment:
Can we make use of `AutoCloseable` and put this in a try-with-resources statement?

```
try(InputStream inputStream = inputFile.newStream()) {
  parse(decompressionEngine.createInputStream(inputStream), eventConsumer);
}
```